modules/generic/src/main/scala/pureconfig/generic/CoproductHint.scala

```diff
@@ -46,6 +48,16 @@ trait CoproductHint[T] {
    * @return `true` if the next class or coproduct option should be tried, `false` otherwise.
    */
   def tryNextOnFail(name: String): Boolean
+
+  /**
+   * Returns a failed `ConfigReader` result scoped into the context of a `ConfigCursor` at the sealed family option,
+   * representing the failure to disambiguate the config value.
+   *
+   * @param cur a `ConfigCursor` at the sealed family option
+   * @return a failed `ConfigReader` result scoped into the context of a `ConfigCursor`.
+   */
+  def failToDisambiguate(cur: ConfigCursor): ConfigReader.Result[CNil] =
```

Comment:
I agree with both points. I prefer asking for `ConfigReaderFailures` because it allows implementations to return more than one failure. I did those changes in 8da8f67.